src/main/kotlin/io/openfuture/state/watcher/ethereum/EthereumBlockProcessor.kt

```diff
@@ -0,0 +1,57 @@
+package io.openfuture.state.watcher.ethereum
+
+import io.openfuture.state.domain.TransactionRequest
+import io.openfuture.state.model.BlockchainType
+import io.openfuture.state.repository.OffsetRepository
+import io.openfuture.state.service.WalletService
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.reactive.asFlow
+import kotlinx.coroutines.reactive.awaitSingle
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Component
+import org.web3j.protocol.Web3j
+import org.web3j.protocol.core.DefaultBlockParameter
+import org.web3j.protocol.core.methods.response.EthBlock
+import java.math.BigInteger
+
+@Component
+class EthereumBlockProcessor(
+        private val offsetRepository: OffsetRepository,
+        private val walletService: WalletService,
+        private val web3j: Web3j
+) {
```

Comment:
Introduce an interface that each processor has to implement in order to enable an additional Blockchain